src/core/shape_inference/include/extract_image_patches_shape_inference.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/extractimagepatches.hpp>
+
+namespace ov {
+namespace op {
+namespace v3 {
+template <class T>
+void shape_infer(const ExtractImagePatches* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
+    const auto& input_shape = input_shapes[0];
+    auto& output_shape = output_shapes[0];
+    output_shape.resize(4);
+    NODE_VALIDATION_CHECK(op, input_shape.rank().compatible(4), "input tensor must be 4D tensor.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_sizes.size() == 2,
+                          "Attribute sizes should be in [size_rows, size_cols] format.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_movement_strides.size() == 2,
+                          "Attribute strides should be in [stride_rows, stride_cols] format.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_movement_strides[0] > 0 && op->m_patch_movement_strides[1] > 0,
+                          "Attribute strides should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_selection_rates.size() == 2,
+                          "Attribute rates should be in [rate_rows, rate_cols] format.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_selection_rates[0] > 0 && op->m_patch_selection_rates[1] > 0,
+                          "Attribute rates should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(
+        op,
+        op->m_padding == PadType::VALID || op->m_padding == PadType::SAME_LOWER || op->m_padding == PadType::SAME_UPPER,
+        "Attribute padding should be in either valid or same_lower or same_upper.");
+    
+    if(input_shape.rank().is_static()) {
+        if (input_shape[1].is_dynamic() || input_shape[2].is_dynamic() || input_shape[3].is_dynamic()) {
+            // Determine Batch
+            output_shape[0] = input_shape[0];
+            return;
+        } else {
```

Comment:
I see that we can fill output shape in parts. This way we will be able to infer the output shape in more cases (For example, when `input_shape[1].is_dynamic()` but `input_shape[2]` and `input_shape[2]` are static).
```
output_shape[0] = input_shape[0];
if (input_shape[1].is_static()) {
    // compute output_shape[1]
}
if (input_shape[2].is_static() && input_shape[3].is_static()) {
    // compute output_shape[2] and output_shape[3]
}
```
Sorry I missed this moment earlier.

Reply:
Great suggestions @v-Golubev , `out_depth` indeed could be calculated in such case. But I think no matter if the `input_shape[1]` is static, we could still calculate it. For example, `input_shape[1]` may be an internal, we can calculated its upper bound & lower bound by `output_shape[1]= input_shape[1] * patch_size[0] * patch_size[1]`

Reply:
@zhangYiIntel I agree, this is how it should be done. Thanks!